### PR TITLE
Add comb function to math module

### PR DESCRIPTION
**Pull request by pingskills**

### CODE REVIEW

File: src/lib/math.js [hunk 1]
```diff
@@ -18,8 +18,40 @@ const $builtinmodule = function (name) {
...
+        if (Sk.ffi.remapToJs(k) === 0) {
+            return new Sk.builtin.int_(1);
+        }
+        if (Sk.ffi.remapToJs(k) > Number.MAX_SAFE_INTEGER) {
+            throw new Sk.builtin.OverflowError("k must not exceed " + Number.MAX_SAFE_INTEGER);
+        }
+
+        const one = new Sk.builtin.int_(1);
+        let tot = n;
+        for (let i = 1; i < k; i++) {
```
Comment (s-cork) [inline src/lib/math.js:48]: Be careful with your types here. `k` is a python object. We shouldn't do this type of comparison between a python object and a javascript object.

File: src/lib/math.js [hunk 2]
```diff
@@ -18,8 +18,40 @@ const $builtinmodule = function (name) {
...
+        if (k < 0) {
+            throw new Sk.builtin.ValueError("k must be a non-negative integer");
+        }
+        if (k > n) {
+            return new Sk.builtin.int_(0);
+        }
+        n = new Sk.builtin.int_(n);
+        k = new Sk.builtin.int_(k);
+        let temp = n.nb$subtract(k);
+        if (Sk.ffi.remapToJs(temp) < Sk.ffi.remapToJs(k)) {
```
Comment (s-cork) [inline src/lib/math.js:36]: you've got a lot of `remapToJs` going on. It's worth noting that `remapToJs` is a slow operation. So doing it several times on the same object is unnecessary.

File: src/lib/math.js [hunk 3]
```diff
@@ -18,8 +18,40 @@ const $builtinmodule = function (name) {
...
+            k = temp
+        }
+        if (Sk.ffi.remapToJs(k) === 0) {
+            return new Sk.builtin.int_(1);
+        }
+        if (Sk.ffi.remapToJs(k) > Number.MAX_SAFE_INTEGER) {
+            throw new Sk.builtin.OverflowError("k must not exceed " + Number.MAX_SAFE_INTEGER);
```
Comment (s-cork) [inline src/lib/math.js:43]: Check your error message against here.

**Review by s-cork — CHANGES_REQUESTED** [review]
Some javascript/python type conversions to consider but otherwise looks good.

File: src/lib/math.js (hunk 1 above)
Comment (pingskills) [reply src/lib/math.js:48]: Ahh, yes. It's so easy to introduce a bug!

File: src/lib/math.js [hunk 4]
```diff
@@ -18,8 +18,41 @@ const $builtinmodule = function (name) {
         return new Sk.builtin.float_(Math.ceil(_x));
     };
 
-    function comb(x, y) {
-        throw new Sk.builtin.NotImplementedError("math.comb() is not yet implemented in Skulpt");
+    function comb(n, k) {
+        n_js = Sk.misceval.asIndexOrThrow(n);
+        k_js = Sk.misceval.asIndexOrThrow(k);
```
Comment (s-cork) [inline src/lib/math.js:23]: `let`

File: src/lib/math.js [hunk 5]
```diff
@@ -18,8 +18,41 @@ const $builtinmodule = function (name) {
...
+        if (k_js > n) {
+            return new Sk.builtin.int_(0);
+        }
+        n = new Sk.builtin.int_(n_js);
+        k = new Sk.builtin.int_(k_js);
+        let temp = n.nb$subtract(k);
+        if (Sk.ffi.remapToJs(temp) < k_js) {
+            k = temp;
+            k_js = Sk.ffi.remapToJs(k);
+        }
```
Comment (s-cork) [inline src/lib/math.js:39]: consider `let temp = Sk.ffi.remapToJs(n.nb$subtract(k));`
We don't use the `k` python object after this line.

**Review by s-cork — CHANGES_REQUESTED** [review]

File: src/lib/math.js (hunk 4 above)
Comment (pingskills) [reply src/lib/math.js:23]: Oh man, is it possible to automatically run eslint over these modules as part of the build process?

File: src/lib/math.js (hunk 5 above)
Comment (pingskills) [reply src/lib/math.js:39]: Yeah, that's much nicer :)

File: src/lib/math.js (hunk 4 above)
Comment (s-cork) [reply src/lib/math.js]: You can probably add an eslint plugin to your editor?

**Review by s-cork — APPROVED** [review]
Another one ready for @skulpt/committers

**Comment by albertjan** [comment]
amazing thanks @pingskills

**Pull request merged**